### PR TITLE
Replace x86 TR::InstOpCode::bad with int3 or ud2 where necessary

### LINKED ISSUE #5374: Deprecate x86 BADIA32Op with either INT3 or UD2
The x86 code generator has long used `BADIA32Op` in place of a breakpoint trap `INT3` instruction.  This is used not only for true diagnostic breakpoints but also in place of an undefined opcode instruction.  However, to make the intent clearer and to align the opcode with the proper Intel mnemonics, uses of `BADIA32Op` should be replaced with either an `INT3` (for a breakpoint trap) or a true `UD2` (for an illegal opcode).

The best way to approach this might be to introduce the `INT3` and `UD2` instructions in OMR first, replace `BADIA32Op` with the appropriate instruction (depending on the context) in downstream projects like OpenJ9, do the same in OMR, and then finally retire `BADIA32Op`.

No functional or performance differences are expected with this change.

### PR DESCRIPTION
**Pull request by BradleyWood**
Closes #5374

### CODE REVIEW

**Comment by BradleyWood** [comment]
Should merge before changes required in OpenJ9 due to the new instructions

File: compiler/x/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -1534,30 +1534,30 @@ void OMR::X86::TreeEvaluator::genArithmeticInstructionsForOverflowCHK(TR::Node *
          case TR::badd:
          case TR::sadd:
          case TR::iadd:
-            addMulAnalyser.integerAddAnalyserWithExplicitOperands(node, operand1, operand2, op, TR::InstOpCode::bad, needsEflags);
+            addMulAnalyser.integerAddAnalyserWithExplicitOperands(node, operand1, operand2, op, TR::InstOpCode::int3, needsEflags);
```
Comment (0xdaryl) [inline compiler/x/codegen/OMRTreeEvaluator.cpp:1537]: For the opcodes passed into the "analyser" functions (not just here, but for all of them), why did you pick `int3` vs. `ud2`?

File: compiler/x/codegen/X86Ops.ins [hunk 2]
```diff
@@ -56,6 +56,14 @@ INSTRUCTION(vgnop, vgnop,
             BINARY(VEX_L___, VEX_vNONE, PREFIX___, REX__, ESCAPE_____, 0x00, 0, ModRM_NONE, Immediate_0),
             PROPERTY0(0),
             PROPERTY1(IA32OpProp1_PseudoOp)),
+INSTRUCTION(ud2, ud2,
+            BINARY(VEX_L___, VEX_vNONE, PREFIX___, REX__, ESCAPE_0F__, 0x0b, 0, ModRM_NONE, Immediate_0),
+            PROPERTY0(0),
+            PROPERTY1(0)),
+INSTRUCTION(int3, int3,
```
Comment (0xdaryl) [inline compiler/x/codegen/X86Ops.ins:63]: The opcode should be upper case but the mnemonic can be lower case like the other instructions.

File: compiler/x/codegen/OMRInstOpCode.enum [hunk 3]
```diff
@@ -26,6 +26,8 @@
 
 #include "compiler/codegen/OMRInstOpCode.enum"
 
+   ud2,
+   int3,
```
Comment (0xdaryl) [inline compiler/x/codegen/OMRInstOpCode.enum:30]: Can you insert these into their somewhat alphabetical place amongst the other instructions?  I only just noticed that the list isn't already perfectly sorted (and we should change that at some point), but if we can still fit these in more or less the right place it make searching the opcodes by a human a lot easier.

**Review by 0xdaryl — CHANGES_REQUESTED** [review]
Can you also shorten the commit title on b63ab2e1808e005663fa8993e4bbec02a652d089 so that it doesn't wrap?  Maybe just drop the "where appropriate" from the title.

File: compiler/x/codegen/OMRTreeEvaluator.cpp (hunk 1 above)
Comment (BradleyWood) [reply compiler/x/codegen/OMRTreeEvaluator.cpp]: Not sure why I did that, but it should be ud2

**Comment by BradleyWood** [comment]
Changes in [force-push](https://github.com/eclipse/omr/compare/b63ab2e1808e005663fa8993e4bbec02a652d089..74d9abcf6642c814af49b55eb1a876172dee4bc8)

**Comment by 0xdaryl** [comment]
Jenkins build xlinux,osx,win

**Comment by 0xdaryl** [comment]
Builds and tests passed on Win64.  Failed due to infra after the build.

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**